src/js/AppStore.js

```diff
@@ -0,0 +1,196 @@
+                }
+
+                var appProperties = {
+                    policyAppID: manifest.appId,
+                    enabled: true
+                }
+
+                let state = store.getState();
+                var appDirEntry = state.appStore.availableApps.find(x => x.policyAppID == manifest.appId);
+
+                if (!appDirEntry) {
+                    console.error('no app found in app store directory with policyAppID ', manifest.appId);
+                    return;
+                }
+
+                store.dispatch(appStoreAddAppPendingSetAppProperties(Object.assign(appDirEntry, { 
+                    policyAppID: manifest.appId,
+                    version: manifest.appVersion,
+                    baseUrl: params.appUrl
+                }), true));
+
+                let addIfExists = (key) => {
+                    if (appDirEntry[key]) {
+                        appProperties[key] = appDirEntry[key];
+                    }
+                }
+
+                addIfExists('nicknames');
+                addIfExists('authToken');
+                addIfExists('transportType');
+                addIfExists('hybridAppPreference');
+
+                BCController.setAppProperties(appProperties);
+            });
+        });
+
+        fileSystemController.sendJSONMessage({
+            method: 'InstallApp',
+            params: {
+                policyAppID: this.state.confirmID,
+                packageUrl: this.state.confirmApp.package.url
+            }
+        });
+
+        this.setState((state, props) => {
+            return {
+                confirmID: null,
+                confirmApp: null
+            }
+        });
+    }
+
+    onSelection(appID) {
+        this.setState((state, props) => {
+            return {
+                confirmID: appID,
+                confirmApp: props.apps.find(app => { return app.policyAppID === appID; })
+            }
+        })
+    }
+
+    componentDidMount() {
+        fetch(flags.AppStoreDirectoryUrl).then((res) => {
+            if (!res.ok) {
+                console.error('could not contact app store server');
+                return [];
+            }
+
+            return res.json();
+        }).then((json) => {
+            store.dispatch(updateAvailableAppStoreApps(json.map((app) => {
+                if ('icon_url' in app) {
+                    app.iconUrl = app.icon_url;
+                    delete app.icon_url;
+                }
+                return app;
+            })));
+        });
+    }
+
+    render() {
+        const themeClass = this.props.theme ? 'dark-theme' : 'light-theme';
+        var modalClass = themeClass + " alertOverlay"
+
+        var pendingInstallApp = this.state.confirmApp ? this.state.confirmApp : {
+            name: null,
+            description: null,
+            iconUrl: null
+        };
+
+        return (
+            <div>
+                <Modal
+                isOpen={!!this.state.confirmID}
```

Comment:
Bang bang?

Reply:
to convert to boolean (react Modal will throw errors in console if it isn't a bool unfortunately)

if you like I could change it to:
```
this.state.confirmID ? true : false
```

Reply:
No that is fine, i just didnt know you could do that